pmd-visualforce/src/main/java/net/sourceforge/pmd/lang/vf/ast/VfTypedNode.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.vf.ast;
+
+import net.sourceforge.pmd.lang.vf.DataType;
+
+/**
+ * Represents a node that displays a piece of data.
+ */
+public interface VfTypedNode extends VfNode {
```

Comment:
I replaced your AbstractVfDataNode with this interface (your abstract class was deprecated). Maybe the name VfDataNode is more appropriate, feel free to rename it